subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -150,6 +150,10 @@ static int wifi_utils_get_all_chans_in_range(uint8_t chan_start,
 		for (i = 0; i < ARRAY_SIZE(valid_5g_chans_20mhz); i++) {
 			if (valid_5g_chans_20mhz[i] == chan_start) {
 				start = true;
+				/* Move the index by 1 to account for the start channel which
+				 * has already been read.
+				 */
+				i++;
```

Comment:
Instead of tweaking loop variable in the middle of the loop, I would prefer that we just use continue here.
So
```
	if (valid_5g_chans_20mhz[i] == chan_start) {
		start = true;
		continue;
	}
```